task-impls/src/view_sync.rs

```diff
@@ -317,15 +320,20 @@ where
                             ));
 
                             let filter = FilterEvent::default();
-                            let _builder =
+                            let builder =
                                 TaskBuilder::<ViewSyncReplicaTaskStateTypes<TYPES, I, A>>::new(
                                     name,
                                 )
                                 .register_event_stream(replica_state.event_stream.clone(), filter)
                                 .await
+                                .register_registry(&mut self.registry.clone())
+                                .await
                                 .register_state(replica_state)
                                 .register_event_handler(replica_handle_event);
-                            // TODO ED Launch task
+
+                            let _view_sync_replica_task =
+                            async_spawn(
+                                ViewSyncReplicaTaskStateTypes::build(builder).launch());
```

Comment:
```suggestion
                            async_spawn(
                                ViewSyncReplicaTaskStateTypes::build(builder).launch()).await;
```
`launch` returns a `BoxFuture` that needs to be awaited in order to be executed